### DISCUSSION TITLE
Implement network ACLs

### DISCUSSION THREAD

**Discussion by orazioedoardo** [pivpn/pivpn#1374] [Feature requests]
We might provide means to restrict access to network resources exposed to VPN clients. Examples:
- Full access
- Internet access, block LAN access
- Lan access only
- PiVPN server only

These should be enforced by the server via the firewall, such that clients cannot override them by changing routes in the client configs.

**Comment by DerDanilo** [comment]
I like the idea. Via unattended file? Via cli/wiptale options?
I currently do selective forwarding via ufw/iptables to restrict client access via vpn. e.g. admin vs user VPN.

**Comment by orazioedoardo** [comment]
Via cli during `pivpn add`, whiptail dialog or text options + command line arguments for non-interactive.

**Comment by albertjimenez** [comment]
any news on this one? Thanks, otherwise, how can we achieve that using other tools? I currently run pivpn on my raspberry pi 4
